clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java

```diff
@@ -81,31 +82,44 @@ public static synchronized void unregisterAppInfo(String prefix, String id, Metr
             if (server.isRegistered(name))
                 server.unregisterMBean(name);
 
-            unregisterMetrics(metrics);
+            unregisterMetrics(metrics, id);
         } catch (JMException e) {
             log.warn("Error unregistering AppInfo mbean", e);
         } finally {
             log.info("App info {} for {} unregistered", prefix, id);
         }
     }
 
-    private static MetricName metricName(Metrics metrics, String name) {
-        return metrics.metricName(name, "app-info", "Metric indicating " + name);
+    private static MetricName metricName(Metrics metrics, String name, Map<String, String> tags) {
+        return metrics.metricName(name, "app-info", "Metric indicating " + name, tags);
     }
 
-    private static void registerMetrics(Metrics metrics, AppInfo appInfo) {
-        if (metrics != null) {
-            metrics.addMetric(metricName(metrics, "version"), (Gauge<String>) (config, now) -> appInfo.getVersion());
-            metrics.addMetric(metricName(metrics, "commit-id"), (Gauge<String>) (config, now) -> appInfo.getCommitId());
-            metrics.addMetric(metricName(metrics, "start-time-ms"), (Gauge<Long>) (config, now) -> appInfo.getStartTimeMs());
+    private static void registerMetrics(Metrics metrics, AppInfo appInfo, String clientId) {
+        if (metrics == null) return;
+        // Most Kafka clients (producer/consumer/admin) set the client-id tag in the metrics config.
+        // Although we don’t explicitly parse client-id here, these metrics are automatically tagged with client-id.
+        metrics.addMetric(metricName(metrics, "version", Map.of()), (Gauge<String>) (config, now) -> appInfo.getVersion());
+        metrics.addMetric(metricName(metrics, "commit-id", Map.of()), (Gauge<String>) (config, now) -> appInfo.getCommitId());
+        metrics.addMetric(metricName(metrics, "start-time-ms", Map.of()), (Gauge<Long>) (config, now) -> appInfo.getStartTimeMs());
+        // MirrorMaker doesn't set client-id tag into the metrics config, so we need to set it here.
```

Comment:
To be precise, the worker also does not set `client-id`
 https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetrics.java#L89